### ISSUE TITLE
lack of latest openssl additions in latest release

### ISSUE THREAD

**Issue by rofl0r** [libressl-portable/portable#115]
compiling stunnel 5.22 against libressl 2.2.1 i get:
```
stunnel-options.o: In function `options_defaults':
options.c:(.text.options_defaults+0x292): undefined reference to `TLS_client_method'
options.c:(.text.options_defaults+0x2a2): undefined reference to `TLS_server_method'
stunnel-verify.o: In function `verify_callback':
verify.c:(.text.verify_callback+0x15a): undefined reference to `X509_check_host'
verify.c:(.text.verify_callback+0x191): undefined reference to `X509_check_email'
verify.c:(.text.verify_callback+0x1c6): undefined reference to `X509_check_ip_asc'
```
at least TLS_client/server_method() seem to be recent additions to the openssl codebase.

**Comment by busterb** [comment]
TLS_client/server_method will be available in the 2.2.2. We'll have to check out the rest post-release.

**Comment by bob-beck** [comment]
X509_check_XXX functions committed to current
